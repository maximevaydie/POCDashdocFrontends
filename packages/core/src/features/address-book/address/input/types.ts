import {Address, ExtractedNewAddress, OriginalAddress} from "dashdoc-utils";
import React from "react";

export type SuggestedAddress = {
    address: Address;
    tooltipContent: string;
    children?: React.ReactNode;
};

export interface OriginalAddressEnriched extends OriginalAddress {
    isLastExtracted?: boolean;
    isLastSuggested?: boolean;
    tooltipContent?: string;
    children?: React.ReactNode;
}

export interface ExtractedNewAddressEnriched extends ExtractedNewAddress {
    isLastExtracted?: boolean;
}

export type SearchAddressResult = {
    options: (OriginalAddressEnriched | ExtractedNewAddressEnriched)[];
    hasMore: boolean;
    additional: {
        page: number;
    };
};
