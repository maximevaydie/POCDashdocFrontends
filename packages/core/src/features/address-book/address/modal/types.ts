import {AddressPayload} from "dashdoc-utils";

import type {AddressTypesOptions} from "../types";

export type AddressForm = AddressPayload & {
    securityProtocolFile?: File | null;
    flowUrl?: string;
    addressTypes: AddressTypesOptions;
};
