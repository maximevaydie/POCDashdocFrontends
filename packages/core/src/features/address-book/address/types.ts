import {SelectOptions} from "@dashdoc/web-ui";

export type AddressType = "is_carrier" | "is_shipper" | "is_destination" | "is_origin";

export function getAllAddressTypes(betterCompanyRolesEnabled: boolean): AddressType[] {
    if (betterCompanyRolesEnabled) {
        return ["is_origin", "is_destination"];
    }
    return ["is_carrier", "is_shipper", "is_destination", "is_origin"];
}

export type AddressTypesOptions = SelectOptions<AddressType>;
