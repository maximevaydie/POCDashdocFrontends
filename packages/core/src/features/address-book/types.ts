import type {SimpleAddress} from "../../types/address";

export type UpdatablePartner = {
    pk: number;
    name: string;
    administrative_address: SimpleAddress;
    notes?: string;
    is_verified?: boolean;
    vat_number?: string;
    invoicing_remote_id?: string;
    can_invite_to: boolean;
};
