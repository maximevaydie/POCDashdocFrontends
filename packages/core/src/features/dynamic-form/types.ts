import {ApiModels} from "../../index";

export type DynamicParameterSpec =
    | ApiModels.DynamicForm.ParameterSpecString
    | ApiModels.DynamicForm.ParameterSpecId
    | ApiModels.DynamicForm.ParameterSpecUid;

export type DynamicFormSpec = ApiModels.DynamicForm.Spec;

export type DynamicParametersSection = {
    name: string;
    description: string;
    parameters: DynamicParameterSpec[];
};

/**
 * The more useful representation of a dynamic form spec to be fed to the react component `DynamicFormComponent`.
 */
export type DynamicForm = {
    sections: DynamicParametersSection[];
};
