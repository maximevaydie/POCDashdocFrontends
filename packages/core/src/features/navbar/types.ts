import {IconNames} from "@dashdoc/web-ui";

export type SidebarChildLink = {
    query: SidebarQuery;
    count?: number;
    countImportant?: boolean;
    countAlert?: boolean;
    link?: string;
    alternativeLinks?: string[];
    label: string;
    icon?: IconNames | null;
    separatorBelow?: boolean;
    testId?: string;
    isNew?: boolean;
};

export type SidebarQuery = {
    tab?: string;
};
