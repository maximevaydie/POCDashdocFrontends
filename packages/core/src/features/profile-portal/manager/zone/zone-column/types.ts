import {Slot} from "types";

export type GridSlotValue = {
    slots: Slot[];
    inOpeningHours: boolean;
    opening: boolean;
    closing: boolean;
};
