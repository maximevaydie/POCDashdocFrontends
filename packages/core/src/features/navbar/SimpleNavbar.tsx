import {BuildConstants, LocaleOption, t} from "@dashdoc/web-core";
import {Box, theme} from "@dashdoc/web-ui";
import {css} from "@emotion/react";
import {User} from "dashdoc-utils";
import React from "react";

import {LanguageSelect} from "./components/LanguageSelect";

const NavbarNoCollapse = css`
    .navbar-header {
        float: left !important;
    }

    .collapse.navbar-collapse {
        display: block !important;
    }
    .navbar-nav > li,
    .navbar-nav {
        float: left !important;
    }

    .navbar-nav.navbar-right:last-child {
        margin-right: -15px !important;
    }

    .navbar-right {
        float: right !important;
    }
`;

const NavbarBrandedLogo = css`
    background-color: white !important;

    .navbar-brand {
        padding: 3px;
        img {
            height: 100%;
            margin: 0;
        }
    }
`;

const NavbarInverseSignature = css`
    background-color: white !important;
`;

interface Props {
    logo?: string | null;
    isSignatureMode?: boolean;
    showLanguageSelector?: boolean;
    onLanguageChange?: (language: string) => void;
    /**
     * This component is used to display the navbar with the new colors
     * but we need to spend time to refactor the navbar and check that it doesn't break anything for others view.
     */
    newColors?: boolean;
    /**
     * When used in Public screens, user is not logged in so we must not display
     * the user dropdown
     */
    forPublicScreen?: boolean;
}

interface UserProps {
    user: Pick<User, "display_name">;
    onLogout: () => void;
}

export function SimpleNavbar(props: Props | (Props & UserProps)) {
    const {logo, newColors = false, forPublicScreen = false} = props;

    return (
        <nav
            className={`navbar navbar-fixed-top ${logo ? "" : "navbar-inverse"}`}
            css={[
                NavbarNoCollapse,
                !!logo && NavbarBrandedLogo,
                props.isSignatureMode && NavbarInverseSignature,
            ]}
            style={
                newColors
                    ? {
                          backgroundColor: theme.colors.blue.dark,
                          borderColor: theme.colors.blue.dark,
                      }
                    : undefined
            }
        >
            <div className="container">
                <div className="navbar-header">
                    {logo ? (
                        <a className="navbar-brand" href="#">
                            <img src={logo} />
                        </a>
                    ) : (
                        <a className="navbar-brand" href="https://www.dashdoc.eu/">
                            <img
                                src={`${BuildConstants.staticUrl}img/logo-dashdoc${
                                    props.isSignatureMode ? "" : "-white"
                                }.png`}
                            />
                        </a>
                    )}
                </div>
                {props.showLanguageSelector && (
                    <div style={{float: "right", width: "70px", paddingTop: "7px"}}>
                        <LanguageSelect
                            onChange={(option: LocaleOption) => {
                                if (props.onLanguageChange && option.value) {
                                    props.onLanguageChange(option.value);
                                }
                            }}
                        />
                    </div>
                )}

                {"user" in props && (
                    <div className="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
                        <ul className="nav navbar-nav navbar-right">
                            {forPublicScreen ? (
                                <Box style={{color: "white"}} mt={3}>
                                    {props.user.display_name}
                                </Box>
                            ) : (
                                <li className="dropdown">
                                    <a
                                        href="#"
                                        className="dropdown-toggle"
                                        data-toggle="dropdown"
                                        role="button"
                                        aria-haspopup="true"
                                        aria-expanded="false"
                                    >
                                        {props.user.display_name}
                                        <span className="caret" />
                                    </a>
                                    <ul className="dropdown-menu">
                                        <li>
                                            <a css={{cursor: "pointer"}} onClick={props.onLogout}>
                                                {t("common.logout")}
                                            </a>
                                        </li>
                                    </ul>
                                </li>
                            )}
                        </ul>
                    </div>
                )}
            </div>
        </nav>
    );
}
