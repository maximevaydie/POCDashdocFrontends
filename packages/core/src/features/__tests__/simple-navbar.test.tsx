import {itRendersCorrectlyWith} from "@dashdoc/web-core/src/testing";
import React from "react";
import {Provider} from "react-redux";
import {createStore, combineReducers} from "redux";

import {accountReducer} from "../../../../../react/Redux/reducers/accountReducer";
import {authReducer} from "../../../../../react/Redux/reducers/authReducer";
import {realtimeReducer} from "../../../../../react/Redux/reducers/realtime.slice";
import {CommonRootState} from "../../../../../react/Redux/types";
import {SimpleNavbar} from "../navbar/SimpleNavbar";

describe("SimpleNavbar", () => {
    const appReducer = combineReducers<CommonRootState>({
        auth: authReducer,
        account: accountReducer,
        realtime: realtimeReducer,
    });
    const store = createStore(appReducer);
    itRendersCorrectlyWith(
        "default",
        <Provider store={store as any}>
            <SimpleNavbar />
        </Provider>
    );
});
