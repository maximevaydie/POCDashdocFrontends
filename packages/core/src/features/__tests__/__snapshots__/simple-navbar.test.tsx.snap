// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleNavbar renders correctly with default 1`] = `
<nav
  className="navbar navbar-fixed-top navbar-inverse"
  css={
    [
      {
        "map": undefined,
        "name": "vr34r6",
        "next": undefined,
        "styles": "
    .navbar-header {
        float: left !important;
    }

    .collapse.navbar-collapse {
        display: block !important;
    }
    .navbar-nav > li,
    .navbar-nav {
        float: left !important;
    }

    .navbar-nav.navbar-right:last-child {
        margin-right: -15px !important;
    }

    .navbar-right {
        float: right !important;
    }
",
        "toString": [Function],
      },
      false,
      undefined,
    ]
  }
>
  <div
    className="container"
  >
    <div
      className="navbar-header"
    >
      <a
        className="navbar-brand"
        href="https://www.dashdoc.eu/"
      >
        <img
          src="undefinedimg/logo-dashdoc-white.png"
        />
      </a>
    </div>
  </div>
</nav>
`;
